### PR TITLE
Fix: error message for non-recognized points in pathfinding

### LINKED ISSUE #9520: Behavior for non-recognized points in the path is not uniform
### What happened?

Depending on which point of the path is not recognized, the behavior is not the same (itinerary or not, error message or not)

### Message :  

If at least one point is not recognized (any point : origin, destination, waypoint), a red message should be shown to the user (the train will be invalid so it is important to show it as soon as possible) : 
- French : "Au moins un des points de passage n'a pas été reconnu"
- English : "At least one of the waypoints could not be recognized"
The red message should be here and look like this (with a different text)
Image

### Itinerary : 

If at least one point is not recognized (any point : origin, destination, waypoint), an itinerary ignoring these unrecognized points should be computed and displayed, except if only one point in the path is recognized. 
In both cases, all the recognized points (even if there is only one) should be shown on the map

### How can we reproduce it (as minimally and precisely as possible)?

Create a case where the origin is not recognized, and another case where a waypoint is not recognized, and observe the different behaviors.

### On which environments the bug occurs?

Local

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

na

### PR DESCRIPTION
**Pull request by achrafmohye**
Closes #9520

### CODE REVIEW

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx [hunk 1]
```diff
@@ -93,7 +97,9 @@ const Pathfinding = ({ pathProperties, setPathProperties }: PathfindingProps) =>
                 <Stop />
               </span>
               <span className="flex-grow-1">
-                {t('pathfindingError', { errorMessage: t(pathfindingState.error) })}
+                {hasInvalidPathStep
+                  ? t('InvalidTrainScheduleStep')
```
Comment (Caracol3) [inline front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:101]: Maybe we should create this to be clearer:
French : "Au moins un des points de passage n'a pas été reconnu"
English : "At least one of the waypoints could not be recognized"

**Comment by Caracol3** [comment]
We now lose the markers and the path on the map. I think we should keep this behavior.

**Comment by Caracol3** [comment]
It works fine for the vias but when just the origin or the destination is invalid we still have the old message of success.
image

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx (hunk 1 above)
Comment (achrafmohye) [reply front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:101]: i'll add this :)

**Comment by Caracol3** [comment]
When  going in itinerary, markers aren't displayed in the map when orign or destination is invalid.
If we had this in usePathding :
```
  useEffect(() => {
    if (origin?.isInvalid || destination?.isInvalid) {
      pathfindingDispatch({
        type: 'PATHFINDING_PARAM_CHANGED',
        params: { origin, destination, rollingStock },
      });
    }
  }, [origin, destination]);
  ```
  
  There is maybe a better way to do it but we have the markers in the map then.

**Review by Caracol3 — APPROVED** [review]
lgtm (tested)

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx [hunk 2]
```diff
@@ -23,16 +26,28 @@ type PathfindingProps = {
...
   } = usePathfinding(setPathProperties, pathProperties);
 
+  useEffect(() => {
+    if (origin?.isInvalid || destination?.isInvalid) {
+      pathfindingDispatch({
+        type: 'PATHFINDING_PARAM_CHANGED',
+        params: { origin, destination, rollingStock },
+      });
+    }
+  }, [origin, destination]);
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx]: Why did you add this useEffect ?

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx [hunk 3]
```diff
@@ -23,13 +26,16 @@ type PathfindingProps = {
...
+  const { getOrigin, getDestination, getPathSteps } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+  const hasInvalidPathStep = pathSteps.some((pathStep) => isPathStepInvalid(pathStep));
   const origin = useSelector(getOrigin, isEqual);
   const destination = useSelector(getDestination, isEqual);
   const { rollingStock } = useStoreDataForRollingStockSelector();
 
   const {
     pathfindingState,
+    pathfindingDispatch,
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx]: You can also remove this
```suggestion
```

Comment (achrafmohye) [reply front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx]: Done :)

**Review by clarani — APPROVED** [review]
LGTM

**Pull request merged**